crucible-llvm/src/Lang/Crucible/LLVM/Intrinsics/Libc.hs

```diff
@@ -487,6 +487,20 @@ callAssert mvar sym (Empty :> _pfn :> _pfile :> _pline :> ptxt ) =
         let err = AssertFailureSimError "Call to assert()" (UTF8.toString txt)
         liftIO $ addFailedAssertion sym err
 
+callExit :: (IsSymInterface sym)
+         => sym
+         -> RegEntry sym (BVType 32)
+         -> OverrideSim p sym ext r args ret (RegValue sym UnitType)
+callExit sym ec = liftIO $
+  do cond <- bvEq sym (regValue ec) =<< bvLit sym knownNat (BV.zero knownNat)
+     -- First, check that the argument is zero. If it is zero, we wish to stop
+     -- execution, without failing.
+     assert sym cond "Call to exit()"
+     -- If we get to this point, the argument is non-zero, so fail.
```

Comment:
With `assert` you state the condition that must be true, and a message to print when it is _not_, so here we'd wany to say "called ext() with nonzero condition".

The abort below is for _all_ cases, 0 and nonzero to kill this thread of execution, so it shouldn't be an `AssertionFailure`.  Maybe we should add a new abort reason for this case, as `InfeasibleBranch` isn't really right either.

Reply:
Thanks for the tips! I've added an `EarlyExit` constructor to `AbortExecReason`. Let me know if you think it deserves to be more descriptive than that.

Reply:
Looks good.